Description: Avoid use of node module cross-spawn
 Use core nodejs function child_process.spawnSync() instead.
Forwarded: no
Author: Jonas Smedegaard <dr@jones.dk>
Last-Update: 2020-04-21
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/lib/util/npm-utils.js
+++ b/lib/util/npm-utils.js
@@ -10,7 +10,7 @@
 //------------------------------------------------------------------------------
 
 const fs = require("fs"),
-    spawn = require("cross-spawn"),
+    spawnSync = require("child_process").spawnSync,
     path = require("path"),
     log = require("./logging");
 
@@ -51,7 +51,7 @@
  */
 function installSyncSaveDev(packages) {
     const packageList = Array.isArray(packages) ? packages : [packages];
-    const npmProcess = spawn.sync("npm", ["i", "--save-dev"].concat(packageList),
+    const npmProcess = spawnSync("npm", ["i", "--save-dev"].concat(packageList),
         { stdio: "inherit" });
     const error = npmProcess.error;
 
@@ -68,7 +68,7 @@
  * @returns {Object} Gotten peerDependencies. Returns null if npm was not found.
  */
 function fetchPeerDependencies(packageName) {
-    const npmProcess = spawn.sync(
+    const npmProcess = spawnSync(
         "npm",
         ["show", "--json", packageName, "peerDependencies"],
         { encoding: "utf8" }
--- a/package.json
+++ b/package.json
@@ -38,7 +38,6 @@
     "@babel/code-frame": "^7.0.0",
     "ajv": "^6.9.1",
     "chalk": "^2.1.0",
-    "cross-spawn": "^6.0.5",
     "debug": "^4.0.1",
     "doctrine": "^3.0.0",
     "eslint-scope": "^4.0.3",
--- a/tests/lib/util/npm-utils.js
+++ b/tests/lib/util/npm-utils.js
@@ -9,7 +9,7 @@
 //------------------------------------------------------------------------------
 
 const assert = require("chai").assert,
-    spawn = require("cross-spawn"),
+    spawnSync = require("child_process").spawnSync,
     sinon = require("sinon"),
     npmUtils = require("../../../lib/util/npm-utils"),
     log = require("../../../lib/util/logging"),
@@ -169,7 +169,7 @@
 
     describe("installSyncSaveDev()", () => {
         it("should invoke npm to install a single desired package", () => {
-            const stub = sandbox.stub(spawn, "sync").returns({ stdout: "" });
+            const stub = sandbox.stub(spawnSync).returns({ stdout: "" });
 
             npmUtils.installSyncSaveDev("desired-package");
             assert(stub.calledOnce);
@@ -179,7 +179,7 @@
         });
 
         it("should accept an array of packages to install", () => {
-            const stub = sandbox.stub(spawn, "sync").returns({ stdout: "" });
+            const stub = sandbox.stub(spawnSync).returns({ stdout: "" });
 
             npmUtils.installSyncSaveDev(["first-package", "second-package"]);
             assert(stub.calledOnce);
@@ -190,7 +190,7 @@
 
         it("should log an error message if npm throws ENOENT error", () => {
             const logErrorStub = sandbox.stub(log, "error");
-            const npmUtilsStub = sandbox.stub(spawn, "sync").returns({ error: { code: "ENOENT" } });
+            const npmUtilsStub = sandbox.stub(spawnSync).returns({ error: { code: "ENOENT" } });
 
             npmUtils.installSyncSaveDev("some-package");
 
@@ -203,7 +203,7 @@
 
     describe("fetchPeerDependencies()", () => {
         it("should execute 'npm show --json <packageName> peerDependencies' command", () => {
-            const stub = sandbox.stub(spawn, "sync").returns({ stdout: "" });
+            const stub = sandbox.stub(spawnSync).returns({ stdout: "" });
 
             npmUtils.fetchPeerDependencies("desired-package");
             assert(stub.calledOnce);
@@ -213,7 +213,7 @@
         });
 
         it("should return null if npm throws ENOENT error", () => {
-            const stub = sandbox.stub(spawn, "sync").returns({ error: { code: "ENOENT" } });
+            const stub = sandbox.stub(spawnSync).returns({ error: { code: "ENOENT" } });
 
             const peerDependencies = npmUtils.fetchPeerDependencies("desired-package");
 
